Python/peephole.c

```diff
@@ -249,17 +256,20 @@ PyCode_Optimize(PyObject *code, PyObject* consts, PyObject *names,
        the peephole optimizer doesn't modify line numbers. */
 
     assert(PyBytes_Check(code));
-    codelen = PyBytes_GET_SIZE(code);
-    assert(codelen % sizeof(_Py_CODEUNIT) == 0);
+    Py_ssize_t codesize = PyBytes_GET_SIZE(code);
+    assert(codesize % sizeof(_Py_CODEUNIT) == 0);
+    Py_ssize_t codelen = codesize / sizeof(_Py_CODEUNIT);
+    if ((size_t)codelen > UINT_MAX) {
```

Comment:
Did you test this on 32-bit platforms? This condition is always false on 32-bit, and this can cause new compiler warnings.

Reply:
No, I didn't. Right now, I have no 32-bit platform.